src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -82,8 +82,8 @@ public ParallelWorkItemDispatcher(int levelOfParallelism)
 
         private void OnStartNonParallelWorkItem(TestWorker worker, WorkItem work)
         {
-            if (work is CompositeWorkItem)
-                SaveQueueState(work);
+            if (work is CompositeWorkItem && work.Test.TypeInfo != null)
+                IsolateQueues(work);
```

Comment:
This is the actual bug fix. It was not caused by what I thought was the problem and described in the issue, although that was also a potential bug.

Specifically, this makes sure we don't create a new set of queues for anything that's not a fixture of some kind. We were doing that for the assembly, when it had `ParallelScope.None` specified.